codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Decoder.java

```diff
@@ -214,17 +222,12 @@ public void decode(InputStream in, HeaderListener headerListener) throws IOExcep
                     } else {
                         nameLength = index;
 
-                        // Disallow empty names -- they cannot be represented in HTTP/1.x
-                        if (nameLength == 0) {
```

Comment:
Hpack allows zero length string headers, but HTTP does not.  The check moved to  DefaultHttp2Headers